godotopenxrkhr/src/main/AndroidManifest.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="org.godotengine.openxrloaders.khr"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <uses-permission android:name="org.khronos.openxr.permission.OPENXR_SYSTEM" />
+
+    <application>
+        <meta-data
+            android:name="org.godotengine.plugin.v1.GodotOpenXRKHR"
+            android:value="org.godotengine.openxrloaders.khr.GodotOpenXRKHR" />
+
+        <activity android:name="com.godot.game.GodotApp" android:exported="true" tools:ignore="MissingClass">
+            <intent-filter>
+                <!-- Mark this activity as using OpenXR. -->
+                <!-- action tag is here to make sure this intent filter is loaded by AOSP. -->
+                <action android:name="org.khronos.openxr.action.MAIN" />
```

Comment:
AOSP filters intent-filters without actions out.

https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/content/pm/PackageParser.java#4575

So we need a 'fake' action here.

Reply:
@rbessems Not sure I understand the reason for this addition, the `GodotApp` activity, which this intent-filter will be merged into already has a defined action, so this fake action should not be needed.

Reply:
Two problems coming together here.

1. intent-filters are not merged. I've tried convincing build tool to do so by adding tools:node="merge" to the intent-filter. However the merged AndroidManifest always gets a 2nd intent-filter.
2. AOSP when loading the AndroidManifest will silently drop any intent filter without an action. This means the category never shows up and the application isn't marked as an OpenXR/AR application.

Reply:
Both issues should be solvable by the option I mentioned below; the `category` addition to the `intent-filter` can be moved to the core Godot's [AndroidManifest generation logic](https://github.com/godotengine/godot/blob/master/platform/android/export/gradle_export_util.cpp), or better yet can be added directly to the [`AndroidManifest` just like we do for Meta's](https://github.com/godotengine/godot/blob/0a9e6e478e3183d7bf3c5cb895c706bc6275d3ea/platform/android/java/app/AndroidManifest.xml#L69) if its inclusion does not have side-effect for regular Godot Android apps running on regular devices (phone, tablet, etc).

Reply:
Sounds good, I'll move this category over there.

Reply:
@m4gr3d adding it to the AndroidManifest like it's done for Meta feels wrong, this tag should really only appear if this an OpenXR activity. 

I've added it to the generator logic, this adds a second intent-filter to the activity which tags the whole activity as an openxr activity. When an application is loaded Android removes any intent-filter without an action, so I add the placeholder action to avoid that.

I've tried to get bundletool to merge the two intent-filters but it is refusing.

Reply:
> adding it to the AndroidManifest like it's done for Meta feels wrong, this tag should really only appear if this an OpenXR activity.

@rbessems  yeah, you're correct. The Meta tag being there by default is also incorrect, so let's clean up the manifest.
This approach was previously used to allow compatibility with Godot's legacy build system for Android, but that build system is deprecated in Godot 4 for XR project, so the Meta category within the AndroidManifest can be removed.

> I've added it to the generator logic, this adds a second intent-filter to the activity which tags the whole activity as an openxr activity. When an application is loaded Android removes any intent-filter without an action, so I add the placeholder action to avoid that.

Sounds good, I'll take a look! Having the two intent-filters is supported and should work as expected, so we can leave it as is.

For reference, according to the doc, `intent-filter` are never merged - https://developer.android.com/studio/build/manage-manifests#merge_conflict_heuristics